AutomatedReview/JenkinsPipelineGuide.md

```diff
@@ -16,6 +16,39 @@ Jenkins server url: https://jenkins.build.o3de.org/
       
 Add sig-build as an approver for all changes to the pipeline.
 
+### Pull Request Considerations
+
+When merging in changes to the Jenkinsfile, it's important to be aware of how Jenkins loads the pipeline files and how it can affect other users. 
+
+Here is a summary of what happens after a user submits a pull request:
+
+1. GitHub creates a ref in the upstream repo based on the source branch of the pull request. This is what Jenkins uses to populate the build node workspace. 
+2. A reviewer/maintainer kicks off the AR to run the builds/tests.
+3. To generate the AR pipelinep, Jenkins merges the contents of the Jenkinsfile from the source branch with the target branch (e.g. development). This ensures incoming changes are running the latest checks and also allows incoming Jenkinsfile updates to be tested. 
+4. The pipeline performs a sparse checkout on the controller to get the files used for pipeline options, scripts to setup the workspace, and other pre-build steps. 
+5. Some scripts from the sparse checkout are stashed so they can be used by the build nodes during the pre/post build steps. 
+6. After the workspace volume is mounted the build node performs the normal checkout of the repo and starts the build/test stages. 
+
+Determining the source of the files used throughout the pipeline, can be confusing. Below is a summary of where Jenkins pulls the files used duing the AR runs:
+
+| File(s) | Description | Source |
+| ------- | --- | ------ |
+| Jenkinsfile | Pipeline definition file | Source branch with latest merged from Target branch |
+| Bootstrap files | Required during pipeline setup steps to load pipeline options, build configs, and setup scripts | Source branch |
+| Workspace contents | Repo contents pulled during a git checkout to run builds and tests | Source branch | 
+
+One important consideration when making an update that uses a new bootstrap file, is that other pull requests will fail if they do not have that file in their branch after your change is merged. This is because the updated Jenkinsfile will be used for their PR and will look for the file. But the file will be missing in their branch (i.e. they have not merged latest yet). 
```

Comment:
Might be clearer to say that **all** script references in the Jenkinsfile depends on the state of the branch, given the table above. If the branch did not pull in latest and the script was merged and referenced in development after the branch creation, it will most likely fail.

Reply:
Good point, it will make it more straight-forward to say all script references. Will update.